processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByQueryEngineV2.java

```diff
@@ -244,38 +257,159 @@ public GroupByEngineIterator(
 
       // Time is the same for every row in the cursor
       this.timestamp = fudgeTimestamp != null ? fudgeTimestamp : cursor.getTime();
+      this.cardinalityFunction = cardinalityFunction;
+      this.allSingleValueDims = allSingleValueDims;
+    }
+
+    private CloseableGrouperIterator<ByteBuffer, Row> initNewDelegate()
+    {
+      final Grouper<ByteBuffer> grouper = newGrouper();
+      grouper.init();
+
+      if (allSingleValueDims) {
+        aggregateSingleValueDims(grouper);
```

Comment:
I had, but lost now. I recall this boosted the performance of ```Simple.A``` of GroupByBenchmark about 20 ~ 30%.